nuxeo-features/nuxeo-platform-audit/nuxeo-platform-audit-mongodb/src/main/java/org/nuxeo/mongodb/audit/MongoDBAuditBackend.java

```diff
@@ -122,7 +124,9 @@ public void onApplicationStarted() {
         MongoDBConnectionService mongoService = Framework.getService(MongoDBConnectionService.class);
         MongoDatabase database = mongoService.getDatabase(AUDIT_DATABASE_ID);
         collection = database.getCollection(collName);
-        // TODO migration ?
+        collection.createIndex(Indexes.ascending(LOG_DOC_UUID)); // query by doc id
```

Comment:
Question @efge please:
why not LOG_PRINCIPAL_NAME or LOG_DOC_LIFE_CYCLE too for instance?

Reply:
I added doc id, date range, and event type because that's what most of the queries I saw were doing. If you think we have other queries that only filter on principal name or lifecycle state, but have no associated filter on doc id, date range or event type, then I'm interested.

Reply:
The idea is that pre-filtering by doc id or date range or event type is usually enough to get efficient results even if we do filter by principal or lifecycle *in addition*.

Reply:
Ok I see, makes sense, thanks